File: src/kyu_4/parseint_reloaded/c/parseint_reloaded_test.c
#include <criterion/criterion.h>

long parse_int (const char* number);

void shouldBe (const char *number, long expected) {
  long result = parse_int (number);
  cr_assert_eq (result, expected, "While parsing '%s' expected %ld, got %ld instead", number, expected, result);
}

Test (parseint, sample) {
    shouldBe ("one", 1);
    shouldBe ("twenty", 20);
    shouldBe ("two hundred and forty-six", 246);
}

File: src/kyu_4/parseint_reloaded/csharp/parseint_reloaded_test.cs
using NUnit.Framework;
public class SolutionTest
{
    [Test]
    public void FixedTests()
    {
        Assert.AreEqual(1, Parser.ParseInt("one"));
        Assert.AreEqual(20, Parser.ParseInt("twenty"));
        Assert.AreEqual(246, Parser.ParseInt("two hundred forty-six"));
    }
}